### PR TITLE
feat(theme): improve color quantization algorithm

### PR DESCRIPTION
**Pull request by DJtheRedstoner**
- fix issues with color quantization returning colors in the wrong order
- remove contrastMode
- use a new algorithm based on HSL color space to derive theme colors from the quantized palette

### CODE REVIEW

File: src/main/kotlin/dev/mediamod/theme/impl/DynamicTheme.kt [hunk 1]
```diff
@@ -15,99 +15,141 @@ class DynamicTheme : Theme.InbuiltTheme("Dynamic") {
...
-            quantizedColors[0].brighter()
+            colors = Colors(
+                background = background,
+                text = text,
+                progressBar = progressBar,
+                progressBarBackground = progressBarBackground,
+                progressBarText = progressBarText
+            )
+        } catch (e: Exception) {
+            e.printStackTrace()
```
Comment (cbyrneee) [inline src/main/kotlin/dev/mediamod/theme/impl/DynamicTheme.kt:56]: Use ``logger.error("An error occurred when doing color quantization: ", e)``

``logger`` is from ``dev.mediamod.utils``

File: src/main/kotlin/dev/mediamod/theme/impl/DynamicTheme.kt [hunk 2]
```diff
@@ -15,99 +15,141 @@ class DynamicTheme : Theme.InbuiltTheme("Dynamic") {
...
-    private fun Color.changeUntil(contrast: Float, against: Color = this): Color {
-        return if (against.luminance > 0.5) {
-            darkenUntil(contrast, against)
-        } else {
-            brightenUntil(contrast, against)
+    private fun Color.contrast(other: Color): Float {
+        var l1 = luminance
+        var l2 = other.luminance
+        if (l2 > l1) {
+            val temp = l1
```
Comment (cbyrneee) [inline src/main/kotlin/dev/mediamod/theme/impl/DynamicTheme.kt:70]: ``temp`` -> ``old``

File: src/main/kotlin/dev/mediamod/utils/ColorQuantizer.kt [hunk 3]
```diff
@@ -315,4 +316,6 @@ object ColorQuantizer {
         fun copy() = VBox(r1, r2, g1, g2, b1, b2, histo)
     }
 
+    class ColorData(val color: Color, val fraction: Float)
+
```
Comment (cbyrneee) [inline src/main/kotlin/dev/mediamod/utils/ColorQuantizer.kt:320]: Remove the newline under:
```kt
class ColorData(val color: Color, val fraction: Float)
```

**Review by caoimhebyrne — CHANGES_REQUESTED** [review]
Just a few very small styling related changes, please amend these to your original commit instead of a fixup.

**Review by caoimhebyrne — APPROVED** [review]
Looks good to me, thank you :)

**Pull request merged**